engine/src/main/java/org/terasology/rendering/opengl/GLSLMaterial.java

```diff
@@ -152,6 +156,43 @@ public void recompile() {
         if (shaderParameters != null) {
             shaderParameters.initialParameters(this);
         }
+
+        //solution due to problems with uniforms.
+        for (Map.Entry<String, Texture> entry : materialData.getTextures().entrySet()) {
+            setTexture(entry.getKey(), entry.getValue());
+        }
+
+        for (Map.Entry<String, Texture> entry : materialData.getTextures().entrySet()) {
+            setTexture(entry.getKey(), entry.getValue());
+        }
+
+        for (Map.Entry<String, Float> entry : materialData.getFloatParams().entrySet()) {
+            setFloat(entry.getKey(), entry.getValue());
+        }
+
+        for (Map.Entry<String, Integer> entry : materialData.getIntegerParams().entrySet()) {
+            setInt(entry.getKey(), entry.getValue());
+        }
+
+        for (Map.Entry<String, float[]> entry : materialData.getFloatArrayParams().entrySet()) {
+            switch (entry.getValue().length) {
```

Comment:
Create a new local variable `value = entry.getValue()` and use that instead. I find that easier to read if used many times in a row. Again, it's a minor issue, really.